src/tests/ftest/util/agent_utils.py

```diff
@@ -65,8 +153,22 @@ def run_agent(basepath, server_list, client_list=None):
     sessions = {}
     user = getpass.getuser()
 
-    # if empty client list, 'self' is effectively client
-    client_list = include_local_host(client_list)
+    # if empty client list, 'test' is effectively client
+    if client_list is None:
+        client_list = include_local_host(client_list)
```

Comment:
The `include_local_host()` method internally handles the `client_list` being `None`, so the conditional check is not required.

Reply:
If without check the client_list is None, it will always add the Host who running the test into client_list.